lib/child-writer/index.js

```diff
@@ -26,6 +26,16 @@ const CONSTANTS = require('./constants')
 const EXIT_CODES = require('../shared/exit-codes')
 const robot = require('../shared/robot')
 
+// There might be multiple Etcher instances running at
+// the same time, therefore we must ensure each IPC
+// server/client has a different name.
+process.env.IPC_SERVER_ID = `etcher-server-${process.pid}`
+process.env.IPC_CLIENT_ID = `etcher-client-${process.pid}`
+
+ipc.config.id = process.env.IPC_SERVER_ID
+ipc.config.socketRoot = CONSTANTS.TMP_DIRECTORY
```

Comment:
Just trying to understand: what is the reason for this change?

Reply:
I've just pulled them out to lowest the scope they need to be in, as they only need to be set once, and logically it didn't make sense to me to have them buried that deep in the handlers.